### DISCUSSION TITLE
Routing along an imported gpx track

### DISCUSSION THREAD

**Discussion by perceptionisfun** [organicmaps/organicmaps#7560] [Q&A]
Let's say I am on a hike and I want to see how far a certain point on the trail is away. When I am routing to that point, the route doesn't follow the gpx track I imported and instead shows the fastest way. Is it possible to route along an imported gpx track?

**Comment by biodranik** [comment]
What does it mean "to route"? An imported route is already visible. You need to see "how far is the point on the track from your current position's projection on the same track", right? How is it implemented in other apps?

**Comment by leonard-slass** [comment]
I think I understand what @perceptionisfun wanted as I just tried the same thing today. I generated a route from A to B using GraphHopper maps. I exported the route as a GPX file. I then imported the file in Organic Maps. I would like to use that route as an offline itinerary (ie. for having direction from A to B using the GPX trace). I could not find anyway to do this.

I hope this is clear.

**Reply by biodranik** [reply]
You can workaround it by adding intermediate stops manually.

**Reply by leonard-slass** [reply]
OK. Thank you.

**Comment by Scoo27** [comment]
i have a similar / same question:
when i import a GPX-File (i think it is a track?) - for example from the [TET Netherlands](https://transeurotrail.org/netherlands/#5.35/52.224/2.869) - i can see the route on the display but i have no button "navigate".
how is it possible to use an imported gpx-file for a turn-by-turn navigation ?
i hav no idea, what @biodranik means when saying "you can add intermediat stops manually".
The track is relatively long and i do not want to add something manually as the route is complete as it is.

Thank you

**Reply by biodranik** [reply]
Track navigation is not implemented yet, see #1360 

You can select the visible track's end point (or any mid point), and build a route. Where it visually won't match the displayed imported track, you can add intermediate stops until a route built by Organic Maps won't match the displayed route.

**Comment by tholehm** [comment]
Same issue here. I planned and saved a route for a road trip. It was saved as "track". Later I wanted to use this "track" to route along, like it had been possible during/after planning. This seems not to be possible. Of course I can create the way points again on that track visible on the map to route finally, but this is unneeded effort. It would be nice to support this "route for track" option.

**Comment by sneko** [comment]
I also think it would be helpful to merge the imported GPX trace with a possible navigation route to have the turn-by-turn navigation.

Some tools are better to plan the trace (for safe cycling...), and Organic Maps would do the job to navigate on mobile.

**Reply by thegrobi** [reply]
Good for planing a route direct in the app (easier than osmand).
But as long, as navigating by an imported gpx-track does not work, the app is useless for me.
